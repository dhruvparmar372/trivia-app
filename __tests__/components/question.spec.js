import React from "react";
import renderer from "react-test-renderer";
import Question from "source/components/question";

const dummyQuestion = {
  id: 1,
  category: "Test",
  question: "Roses are only red&quot;",
  correct_answer: "False",
  incorrect_answers: ["True"]
};

describe("question component", () => {
  test("renders question", () => {
    expect(
      renderer.create(
        <Question
          question={dummyQuestion}
          ignoreAnswerShuffle
          onAnswer={jest.fn}
        />
      )
    ).toMatchSnapshot();
  });
});
