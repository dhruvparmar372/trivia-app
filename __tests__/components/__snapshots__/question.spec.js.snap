// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`question component renders question 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "System",
          "fontSize": 18,
        },
        Object {
          "color": "#eff0f4",
          "marginBottom": 15,
        },
      ]
    }
  >
    Test
  </Text>
  <View
    style={
      Object {
        "borderColor": "#eff0f4",
        "borderRadius": 2,
        "borderWidth": 1,
        "marginBottom": 24,
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "System",
            "fontSize": 18,
          },
          Object {
            "color": "#eff0f4",
            "fontSize": 29,
          },
        ]
      }
    >
      Roses are only red"
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#74dbef",
        "borderRadius": 2,
        "marginTop": 10,
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "System",
            "fontSize": 18,
          },
          Object {
            "color": "#264e86",
            "textAlign": "center",
          },
        ]
      }
    >
      True
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#74dbef",
        "borderRadius": 2,
        "marginTop": 10,
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "System",
            "fontSize": 18,
          },
          Object {
            "color": "#264e86",
            "textAlign": "center",
          },
        ]
      }
    >
      False
    </Text>
  </View>
</View>
`;
